c/cmake_modules/AdbcDefines.cmake

```diff
@@ -31,6 +31,11 @@ set(ADBC_VERSION_MAJOR "9")
 set(ADBC_VERSION_MINOR "0")
 set(ADBC_VERSION_PATCH "0")
 
+set(CXX_STANDARD_REQUIRED ON)
+if(NOT "${CMAKE_CXX_STANDARD}")
+  set(CMAKE_CXX_STANDARD 17)
+endif()
+
```

Comment:
Could perhaps also add `target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_17)` somewhere, though I'm not sure where.

Reply:
Ah, thanks. Added it to the `add_arrow_lib` function for now (though we really should clean up the CMake here)